### PR TITLE
Make Cloud SDK validation consistent

### LINKED ISSUE #1203: Consistent Cloud SDK and App Engine components validation
[`DeployPreferencesDialog`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java#L150)

[`CloudSdkPreferenceArea`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java#L163)

[`AppEngineFacetInstallDelegate`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFacetInstallDelegate.java#L43)

[`LocalAppEngineServerLaunchConfigurationDelegate`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java#L85)

[`StandardProjectWizard`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/StandardProjectWizard.java#L64)

[`MavenArchetypeProjectWizard`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenArchetypeProjectWizard.java#L65)

[`CloudSdkContextFunction`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkContextFunction.java#L76)

### PR DESCRIPTION
**Pull request by chanseokoh**
About #1203.

There could be some subtleties, so need to look at the changes closely.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java [hunk 1]
```diff
@@ -157,11 +161,6 @@ private IStatus validateAppEngineJavaComponents()  {
         Status status = new Status(IStatus.ERROR,
             "com.google.cloud.tools.eclipse.appengine.deploy.ui", detailMessage);
         return status;
-    } catch (AppEngineException ex) {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java]: By taking this out, you lost the check for a version out of date.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/.settings/.jsdtscope [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/.settings/.jsdtscope]: I'm not sure why this is here?

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java (hunk 1 above)
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java]: `CloudSdkOutOfDateException` is being caught right above. (Not shown unless you expand this GitHub diff.)

**Comment by chanseokoh** [comment]
@briandealwis I like the idea. Error UI messages are all different, so we need to think about that. If we decide to do so, I suggest doing it in a separate PR.

**Pull request merged**